nixos/modules/services/hardware/interception-tools.nix

```diff
@@ -0,0 +1,60 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.interception-tools;
+in {
+  options.services.interception-tools = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = "Whether to enable the interception tools service.";
+    };
+
+    plugins = mkOption {
+      type = types.listOf types.package;
+      default = [ pkgs.interception-tools-plugins.caps2esc ];
+      description = ''
+        A list of interception tools plugins that will be made available to use
+        inside the udevmon configuration.
+      '';
+    };
+
+    udevmonConfig = mkOption {
+      type = types.either types.str types.path;
+      default = ''
+        - JOB: "intercept -g $DEVNODE | caps2esc | uinput -d $DEVNODE"
+          DEVICE:
+            EVENTS:
+              EV_KEY: [KEY_CAPSLOCK, KEY_ESC]
+      '';
+      example = ''
+        - JOB: "intercept -g $DEVNODE | y2z | x2y | uinput -d $DEVNODE"
+          DEVICE:
+            EVENTS:
+              EV_KEY: [KEY_X, KEY_Y]
+      '';
+      description = ''
+        String of udevmon YAML configuration, or path to a udevmon YAML
+        configuration file.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.interception-tools = {
+      description = ''
+        Minimal composable infrastructure on top of libudev and libevdev
+      '';
+      path = [ pkgs.bash pkgs.interception-tools ] ++ cfg.plugins;
+      script = ''
+        nice -n -20 udevmon -c \
```

Comment:
The `Nice=` property of systemd can do the same. Then `script` can be replaced by `service.ExecStart`.

Reply:
Why is it actually necessary to change the nice level at all? Shouldn't user input be processed as fast as possible?

Reply:
Thanks, updated. I didn't know about `Nice`.

And yeah if you check the manpage for `systemd.exec` and search for nice, it says that -20 is the highest priority. But I just got this from the [package readme](https://gitlab.com/interception/linux/tools#execution), where it says that it increases the `udevmon` priority since it'll be responsible for a vital input device, just to make sure it stays "responsible".